packages/roosterjs-content-model-dom/lib/modelApi/editing/normalizeTable.ts

```diff
@@ -52,12 +54,17 @@ export function normalizeTable(
             const cell = mutateBlock(readonlyCell);
 
             if (cell.blocks.length == 0) {
-                const format = cell.format.textColor
-                    ? {
-                          ...defaultSegmentFormat,
-                          textColor: cell.format.textColor,
-                      }
-                    : defaultSegmentFormat;
+                const cellFormat = { ...cell.format };
+                // Remove all non-segment related properties from cell format
+                getObjectKeys(cellFormat).forEach(key => {
```

Comment:
This will delete format that exists in ContentModelTableCellFormat but not in ContentModelSegmentFormat.

Actually ContentModelTableCellFormat  is not a sub class of ContentModelSegmentFormat, so better not delete like this

Reply:
That is the purpose of the function. Why would ContentModelTableCellFormat not being a sub class of ContentModelSegmentFormat affect this? This change adds the properties of ContentModelSegmentFormat to ContentModelTableCellFormat.

Reply:
We expect segment format to be applied to segment only. For example, if we want to apply font-family, do it on text but not on table cell:

Do:
```html
<td><span style="font-family: xxx">text</span></td>
```

Do not do:
```html
<td style="font-family: xxx">text</td>
```

So that each level has its only responsibility. And when we process table, no need to handle those segment formats. It is not table's responsibility.

That is why ContentModelTableCellFormat does not contain ContentModelSegmentFormat.

It only contains text color, given that we need to apply a default text color when set table cell's shade color. It is just a default color and it can always be overwritten by color from segment.